src/AGENTS.md

```diff
@@ -295,6 +326,41 @@ class MultiProducer:
         return KafkaProducer(build_kafka_configuration(default_config=config))
 ```
 
+## Code Comments
```

Comment:
we previously had this in our instructions, and I did not observe it to help -- could you try a quick sanity check seeing if this actually improves in reducing comments?

(I think we should try to be conservative as to not pollute the context window)